modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/Command.java

```diff
@@ -51,8 +52,13 @@ public static GridClient startClient(GridClientConfiguration clientCfg) throws E
         GridClient client = GridClientFactory.start(clientCfg);
 
         // If connection is unsuccessful, fail before doing any operations:
-        if (!client.connected())
-            client.throwLastError();
+        if (!client.connected()) {
+            GridClientException lastErr = client.checkLastError();
+
+            client.close();
```

Comment:
What happens if close() throws another exception? It seems that lastErr will be lost. Can we try-catch around client.close() and put the exception to suppressed if it happens?